types/coin.go

```diff
@@ -11,14 +13,59 @@ type Coin struct {
 }
 
 func (coin Coin) String() string {
-	return fmt.Sprintf("(%v %v)",
-		coin.Denom, coin.Amount)
+	return fmt.Sprintf("%v%v", coin.Amount, coin.Denom)
+}
+
+//regex codes for extracting coins from string
+var reDenom = regexp.MustCompile("([^\\d\\W]+)")
+var reAmt = regexp.MustCompile("(\\d+)")
+
+func ParseCoin(str string) (Coin, error) {
+
+	var coin Coin
+
+	if len(str) > 0 {
+		amt, err := strconv.Atoi(reAmt.FindString(str))
+		if err != nil {
+			return coin, err
+		}
+		denom := reDenom.FindString(str)
+		coin = Coin{denom, int64(amt)}
+	}
+
+	return coin, nil
 }
 
 //----------------------------------------
 
 type Coins []Coin
 
+func (coins Coins) String() string {
+	out := ""
+	for _, coin := range coins {
+		out += fmt.Sprintf("%v,", coin.String())
```

Comment:
but we truncate it in return one line below... I'm going to add a nil check though at the beginning to avoid panics